src/router/routing.rs

```diff
@@ -0,0 +1,82 @@
+use super::{DevAddrFilter, EuiFilter};
+use crate::{KeyedUri, PublicKey, Result};
+use helium_proto::{routing_information::Data as RoutingData, RoutingInformation};
+use slog::{warn, Logger};
+use std::{convert::TryFrom, sync::Arc};
+
+#[derive(Clone, Debug)]
+pub struct Routing {
+    pub(crate) oui: u32,
+    pub(crate) uris: Vec<KeyedUri>,
+    filters: Vec<EuiFilter>,
+    subnets: Vec<DevAddrFilter>,
+}
+
+impl Routing {
+    pub fn contains_uri(&self, uri: &KeyedUri) -> bool {
+        self.uris.iter().any(|keyed_uri| keyed_uri == uri)
+    }
+
+    pub fn matches_routing_info(&self, routing_info: &Option<RoutingInformation>) -> bool {
```

Comment:
Is this not called anywhere?

Reply:
Called here https://github.com/helium/gateway-rs/blob/madninja/router_support/src/router/dispatcher.rs#L292